sdk/core/azure-core/tests/async_tests/test_base_polling_async.py

```diff
@@ -283,6 +283,36 @@ async def send(request, **kwargs):
     assert result["location_result"] == True
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "http_request,http_response", request_and_responses_product(ASYNCIO_REQUESTS_TRANSPORT_RESPONSES)
+)
+async def test_post_direct_success(async_pipeline_client_builder, deserialization_cb, http_request, http_response):
+
+    # ResourceLocation
+
+    # The initial response contains both Location and Operation-Location, a 202 and a success body
+    initial_response = TestBasePolling.mock_send(
+        http_request,
+        http_response,
+        "POST",
+        202,
+        {
+            "operation-location": "http://example.org/async_monitor",
+        },
+        {"status": "succeeded"},
```

Comment:
Let's add some tests for failure terminal states as well....

Reply:
Added.